### DISCUSSION TITLE
[Question] Issue with `MatplotlibChart` and Specified `xlim` and `ylim`

### DISCUSSION THREAD

**Discussion by yuheiyta** [flet-dev/flet#1836] [Q&A] [answered]
### Question

I've encountered an issue while using `MatplotlibChart` in conjunction with setting `xlim` and `ylim` in a `matplotlib.pyplot.figure` object. It seems that the resultant plot extends beyond the specified x and y limits.
Any insights or solutions regarding this problem would be greatly appreciated. Thank you!
<img width="797" alt="flet_screenshot" src="https://github.com/flet-dev/flet/assets/50651967/29fa6bd8-b21e-4aa5-b749-a211a4bf9418">

### Code sample

```python
import flet as ft
from flet import Page
from flet.matplotlib_chart import MatplotlibChart
import matplotlib
import matplotlib.pyplot as plt
import numpy as np

matplotlib.use("Agg")

def main(page: Page):
    
    fig = plt.figure()
    ax=fig.add_subplot()
    x=np.arange(0,100)
    y=x*2
    ax.plot(x,y)
    ax.set_xlim(10,)
    ax.set_ylim(30,)
    mplc = MatplotlibChart(figure=fig)

    page.add(mplc)
ft.app(target=main)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by tehphilb** [comment]
From the Matplotlib docs:
```
plt.xlim(0,0) # Lower Limit, Upper Limit
plt.ylim(0,0) # Lower Limit, Upper Limit
```

Your example:
```
import flet as ft
from flet import Page
from flet.matplotlib_chart import MatplotlibChart
import matplotlib
import matplotlib.pyplot as plt
import numpy as np

matplotlib.use("Agg")

def main(page: Page):
    
    fig = plt.figure()
    ax=fig.add_subplot()
    x=np.arange(0,100)
    y=x*2
    ax.plot(x,y)
    ax.set_xlim(-10,)
    ax.set_ylim(-30,)
    mplc = MatplotlibChart(figure=fig)

    page.add(mplc)
ft.app(target=main)
```

![2023-09-14 13_40_47](https://github.com/flet-dev/flet/assets/49588177/9d76d99f-4035-4cb3-a5aa-42cbc78fe903)

**Reply by yuheiyta** [reply]
My point is that when showing with matplotlib, plots outside the range are hidden, whereas with `MatplotlibChart` the plots outside the range are still visible for some reason.

**Reply by tehphilb** [reply, marked as answer]
Now I got it.

It seems to be an issue with the clip_on attribute. https://matplotlib.org/3.2.2/api/_as_gen/matplotlib.axes.Axes.set_clip_on.html

Note that clip_on=True is actually the default behavior for the plotting functions in Matplotlib. It's seems possible that Flet, how it renders the Matplotlib charts, and the SVG representation may not respect this setting, so far.

![2023-09-15 11_08_10](https://github.com/flet-dev/flet/assets/49588177/f0c2a728-0cb7-49ff-9348-e8db556fbaaf)

**Reply by yuheiyta** [reply]
Thanks! I found some issues reported in `matplotlib`. Seemingly that this issue is not caused by `matplotlib` but svg handling part https://github.com/matplotlib/matplotlib/pull/2423#issuecomment-343800856.
